Bugfix/s3 c 3319 notification constants correction

naren-rajendran opened the pull request:
Bucket notification constants correction.

- Removed redundant `zkBucketNotificationPath` constant.
- Corrected `nameFilter` prefix and suffix values.

bert-e commented:
# Hello naren-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/backbeat/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3319-notification-constants-correction` into
`development/7.8`
* [w/8.1/bugfix/S3C-3319-notification-constants-correction](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/S3C-3319-notification-constants-correction) will be merged into `development/8.1` (pull request #1026)
* [w/8.2/bugfix/S3C-3319-notification-constants-correction](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/bugfix/S3C-3319-notification-constants-correction) will be merged into `development/8.2` (pull request #1027)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

bert-e commented:
# History mismatch



Merge commit #44c46a8722e947775f4cb0e93d57e752f932970f on the integration branch
`w/8.1/bugfix/S3C-3319-notification-constants-correction` is merging a branch which is neither the current
branch `bugfix/S3C-3319-notification-constants-correction` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/S3C-3319-notification-constants-correction` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

jonathan-gramain commented on extensions/notification/utils/config.js:
```diff
@@ -32,11 +32,13 @@ function configArrayToMap(bnConfigs) {
-            return key.startsWith(rule.value);
+        const { name, value } = rule;
+        const { prefix, suffix } = constants.nameFilter;
+        if (name.toLowerCase() === prefix.toLowerCase()) {
```
Is this correct if both a prefix and a suffix are given? Meaning the fact that we return from the function with the first check's result.

naren-rajendran replied:
`filterRules` is an array with either `prefix` or `suffix` and it is checked for all the rules. `Array.every` is used on the array.

Commit:
bf: S3C-3319 Notification constants correction

- Removed redundant `zkBucketNotificationPath` constant
- Corrected `nameFilter` prefix and suffix values

bert-e commented:
# History mismatch



Merge commit #6f1140d3caf3b64cc65451219df6294b8be0688a on the integration branch
`w/8.1/bugfix/S3C-3319-notification-constants-correction` is merging a branch which is neither the current
branch `bugfix/S3C-3319-notification-constants-correction` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/S3C-3319-notification-constants-correction` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

naren-rajendran commented:
@bert-e approve

bert-e commented:
# History mismatch



Merge commit #6f1140d3caf3b64cc65451219df6294b8be0688a on the integration branch
`w/8.1/bugfix/S3C-3319-notification-constants-correction` is merging a branch which is neither the current
branch `bugfix/S3C-3319-notification-constants-correction` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/S3C-3319-notification-constants-correction` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

naren-rajendran commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3319-notification-constants-correction` into
`development/7.8`
* [w/8.1/bugfix/S3C-3319-notification-constants-correction](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/S3C-3319-notification-constants-correction) will be merged into `development/8.1` (pull request #1036)
* [w/8.2/bugfix/S3C-3319-notification-constants-correction](https://github.com/scality/backbeat/compare/development/8.2...w/8.2/bugfix/S3C-3319-notification-constants-correction) will be merged into `development/8.2` (pull request #1037)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/backbeat/commit/0b6422116d82f99611bf63ece28ba96a34ae52cb) did not succeed in branch w/8.2/bugfix/S3C-3319-notification-constants-correction.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.8`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`



The following branches will **NOT** be impacted:

* `development/7.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.8`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`



The following branches have **NOT** changed:

* `development/7.4`



Please check the status of the associated issue S3C-3319.

Goodbye naren-scality.
